Libraries/libbom/Tools/lsbom.cpp

```diff
@@ -114,67 +114,91 @@ Options::
 {
 }
 
+static ext::optional<std::vector<Options::PrintItem>>
+ParsePrintFormat(std::string const &print)
+{
+    auto format = std::vector<Options::PrintItem>();
+
+    for (char c : print) {
+        switch (c) {
```

Comment:
Hmm...is it possible you do do something more clever, like create a map for characters -> options, then just reference that?